### PR TITLE
react-app-rewired support react-scripts more than 2.1.2 version

### PR DESCRIPTION
**Pull request by vdfor**

### CODE REVIEW

**Comment by AndyOGo** [comment]
@vdfor 
Just switch to CRACO JS (crafted for CRA V2)
https://github.com/sharegate/craco

**Comment by marceloadsj** [comment]
Maybe CRACO is broken too, haha! :-(

I think it's broken because CRA 2.1.2 merged `webpack.config.dev` with `webpack.config.prod`,using condition to check dev/prod envs...

https://github.com/facebook/create-react-app/pull/5722

Checked CRACO, it's using old approach too!
https://github.com/sharegate/craco/blob/master/packages/craco/lib/cra.js#L25

Can someone confirm this?

**Comment by AndyOGo** [comment]
@vdfor
@marceloadsj 
What a bummer, unfortunately this repo is not maintained any more.
So it may be better to provide this PR and issue description at CRACO JS

**Comment by timarney** [comment]
If some people want to confirm this PR works I will merge.

**Comment by vdfor** [comment]
> @vdfor
> @marceloadsj
> What a bummer, unfortunately this repo is not maintained any more.
> So it may be easier to provide this PR and issue description at CRACO JS

https://github.com/sharegate/craco/pull/62
This pr will fix it at CRACO JS

File: packages/react-app-rewired/scripts/build.js [hunk 1]
```diff
@@ -2,12 +2,16 @@ process.env.NODE_ENV = 'production';
 
 const paths = require('./utils/paths');
 const overrides = require('../config-overrides');
-const webpackConfigPath = paths.scriptVersion + "/config/webpack.config.prod";
+const scriptPkg = require(paths.scriptVersion + "/package.json");
+
+const isOldScript = !(scriptPkg && scriptPkg.version >= '2.1.2');
```
[the remark this answers is not in the document]
Comment (hckhanh) [reply packages/react-app-rewired/scripts/build.js]: We need to use the module to compare version
https://github.com/omichelsen/compare-versions
or
https://github.com/npm/node-semver

**Comment by tangdw** [comment]
CRACO is broken too, switch to https://github.com/rescripts/rescripts

**Comment by timarney** [comment]
Closing this as this one has been merged.
https://github.com/timarney/react-app-rewired/pull/346

**Pull request closed without merging**